vips/process.go

```diff
@@ -562,45 +590,49 @@ func supportedSaveFormat(format ImageType) ImageType {
 	return ImageTypeJPEG
 }
 
-func (v *Processor) export(image *Image, format ImageType, quality int) ([]byte, error) {
-	switch format {
+func (v *Processor) export(image *Image, params *ExportParams) ([]byte, error) {
+	switch params.format {
 	case ImageTypePNG:
 		opts := NewPngExportParams()
+		opts.Quality = params.Quality
+		opts.Palette = params.Palette
+		opts.Bitdepth = params.Bitdepth
+		opts.Compression = params.Compression
```

Comment:
Add a `if` condition to only populate the options if being set.
In Go `int` is default at `0` and `bool` is default at `false`

https://github.com/cshum/imagor/blob/master/vips/foreign.go#L48 
As you can see the default compression is 6. Should only override `if compression > 0`
This avoid default params being overridden, resulting breaking changes.